advanced_source/super_resolution_with_onnxruntime.py

```diff
@@ -0,0 +1,285 @@
+        init.orthogonal_(self.conv4.weight)
+
+# Create the super-resolution model by using the above model definition.
+torch_model = SuperResolutionNet(upscale_factor=3)
+
+
+######################################################################
+# Ordinarily, you would now train this model; however, for this tutorial,
+# we will instead download some pre-trained weights. Note that this model
+# was not trained fully for good accuracy and is used here for
+# demonstration purposes only.
+#
+
+# Load pretrained model weights
+model_url = 'https://s3.amazonaws.com/pytorch/test_data/export/superres_epoch100-44c6958e.pth'
+batch_size = 1    # just a random number
+
+# Initialize model with the pretrained weights
+map_location = lambda storage, loc: storage
+if torch.cuda.is_available():
+    map_location = None
+torch_model.load_state_dict(model_zoo.load_url(model_url, map_location=map_location))
+
+# set the train mode to false since we will only run the forward pass.
+torch_model.train(False)
+
+
+######################################################################
+# Exporting a model in PyTorch works via tracing or scripting. This
+# tutorial will use as an example a model exported by tracing. 
+# To export a model, you call the ``torch.onnx.export()`` function.
+# This will execute the model, recording a trace of what operators
+# are used to compute the outputs.
+# Because ``_export`` runs the model, we need to provide an input
+# tensor ``x``. The values in this can be random as long as it is the
+# right type and size.
+#
+# To learn more details about PyTorch's export interface, check out the
+# `torch.onnx documentation <https://pytorch.org/docs/master/onnx.html>`__.
+#
+
+# Input to the model
+x = torch.randn(batch_size, 1, 224, 224, requires_grad=True)
+
+# Export the model
+torch_out = torch.onnx._export(torch_model,               # model being run
+                               x,                         # model input (or a tuple for multiple inputs)
+                               "super_resolution.onnx",   # where to save the model (can be a file or file-like object)
+                               export_params=True,        # store the trained parameter weights inside the model file
+                               opset_version=10,          # the onnx version to export the model to
+                               do_constant_folding=True,  # wether to execute constant folding for optimization
+                               input_names = ['input'],   # the model's input names
+                               output_names = ['output'], # the model's output names
+                               dynamic_axes={'input' : {0 : 'batch_size'},    # variable lenght axes
+                                             'output' : {0 : 'batch_size'}})
+
+######################################################################
+# ``torch_out`` is the output after executing the model. Normally you can
```

Comment:
After switching to torch.onnx.export, you can call the model with the same input and get example output, which can be used to verify the exported model.